RegistrationForm

MaBelenBecerra opened the pull request:
Registro de restaurante con los campos de: Nombre de restaurante, dirección, descripción, telefono, hora de cierre y hora de apertura. Sin conexion al Supabase
![image](https://github.com/user-attachments/assets/583fe1d5-d9c7-415c-831a-aeb27b102d76)

MaBelenBecerra committed:
RegistrationForm

Registro de restaurante con los campos de: Nombre de restaurante, dirección, descripción, telefono, hora de cierre y hora de apertura

henrrycoronado commented on OrderNow-React/src/pages/Home.jsx:
```diff
@@ -1,13 +1,14 @@
       <div className="flex flex-col items-center ">
         <h1 className="text-4xl font-bold mb-6">HOME PAGE</h1>
         <BotonRestaurantes />
+        <RegistrationForm />
```
el llamado a la pagina deberia estar en una ruta diferente al "Home" , debiendo definirse eso en la hoja "App.jsx" donde estan todas las rutas definidas

RaidenMT replied:
Siguiendo la anterior lógica por ahora se cierra, pero muy buena observación Henry

henrrycoronado commented on OrderNow-React/src/components/RegistrationForm/RegistrationForm.jsx:
El nombre del componente debe ser mas explicito que va dirigido a el restaurante, ya que existe en este directorio otro componente "RegistroForm" para el registro de los clientes

RaidenMT replied:
Ya se movió de proyecto, por lo que se podrá cerrar este comentario.

henrrycoronado commented on OrderNow-React/src/pages/Home.jsx:
```diff
@@ -1,13 +1,14 @@
 import React from 'react';
 import BotonRestaurantes from '../components/BotonRestaurantes'; 
-
+import RegistrationForm from '../components/RegistrationForm/RegistrationForm';
```
no se deberia llamar a el componente directamente si se hara una vista aparte del registro de restauurante, el componente debe ser llamado y estar dentro de una pagina independiente y propia, y esa pagina es la que debe ser invocada en las rutas definida en el enrutador en "App.jsx"

RaidenMT replied:
Si bien Henry tiene toda la razón, por ahora lo dejaremos así porque es parte del alcance de esta HU, por temas de tiempo.

henrrycoronado reviewed (commented):
revisar observaciones

RaidenMT commented on OrderNow-React/package.json:
Mover a proyecto OrderNow Dashboard

MaBelenBecerra replied:
Hecho

RaidenMT commented on OrderNow-React/src/components/RegistrationForm/RegistrationForm.jsx:
Mover a proyecto OrderNow Dashboard

RaidenMT commented on OrderNow-React/src/components/RegistrationForm/RegistrationForm.jsx:
```diff
@@ -0,0 +1,206 @@
+
+    if (!formData.phone.trim()) {
+      newErrors.phone = 'El teléfono es requerido';
+    } else if (!/^\+?[\d\s-]{8,}$/.test(formData.phone.trim())) {
```
Convertir en variable el regex

RaidenMT commented on OrderNow-React/src/components/RegistrationForm/RegistrationForm.jsx:
```diff
@@ -0,0 +1,206 @@
+
+    if (!formData.closingTime) {
+      newErrors.closingTime = 'El horario de cierre es requerido';
+    } else if (formData.openingTime && formData.closingTime <= formData.openingTime) {
```
Para que closingTime sea posterior al openingTime debe ser solo "mayor a"

RaidenMT replied:
@MaBelenBecerra Falta resolver este caso

MaBelenBecerra replied:
Não não

RaidenMT commented on OrderNow-React/src/components/RegistrationForm/RegistrationForm.jsx:
```diff
@@ -0,0 +1,206 @@
+    }
+
+    setErrors(newErrors);
+    return Object.keys(newErrors).length === 0;
```
¿Me puedes explicar esta línea de código? Qué es lo que hará

MaBelenBecerra replied:
Guarda los errores en el estado y devuelve true solo si el formulario está válido ( es decir sin errores :))

RaidenMT commented on OrderNow-React/src/components/RegistrationForm/RegistrationForm.jsx:
```diff
@@ -0,0 +1,206 @@
+      setSubmitted(true);
+
+      setTimeout(() => {
+        setFormData({
```
¿Por qué va en un timeout?

RaidenMT commented on OrderNow-React/src/components/RegistrationForm/RegistrationForm.jsx:
```diff
@@ -0,0 +1,206 @@
+      setSubmitted(true);
+
+      setTimeout(() => {
+        setFormData({
```
¿Qué hace este código?

MaBelenBecerra replied:
Ya nada porque lo elimine ;)

RaidenMT reviewed (changes requested):
Modificar por favor los cambios solicitados

MaBelenBecerra committed:
Correciones

Converti en variable el phoneRegex
Quite el SetTimeout
Instale tailwind
Movi el archivo al Dashboard
Cree una carpeta de components y movi el archivo de RegistrationForm al archivo de components

RaidenMT commented on OrderNow-Dashboard/src/App.jsx:
```diff
@@ -2,33 +2,14 @@ import { useState } from 'react'
-      </p>
-    </>
+
+    <RegistrationForm/>
```
En otra PR se solicitará mover de página. Dejo esto solamente como comentario

MaBelenBecerra committed:
Merge branch 'DashboardElMilagro'
